pkgs/by-name/ge/gerbolyze/package.nix

```diff
@@ -0,0 +1,92 @@
+{ lib
+, stdenv
+, pythonPackages
+, fetchFromGitHub
+, resvg
+}:
+
+let
+  pname = "gerbolyze";
+  version = "v3.1.7";
+  src = fetchFromGitHub {
+    owner = "jaseg";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-0iTelSlUJUafclRowwsUAoO44nc/AXaOKXnZKfKOIaE=";
+    fetchSubmodules = true;
+  };
+
+  svg-flatten = stdenv.mkDerivation rec {
+    inherit version src;
+    pname = "svg-flatten";
+
+    sourceRoot = "source/svg-flatten";
+
+    postPatch = ''
+      substituteInPlace Makefile \
+        --replace '$(INSTALL) $(BUILDDIR)/$(BINARY) $(PREFIX)/bin' \
+        '$(INSTALL) $(BUILDDIR)/$(BINARY) $(PREFIX)/bin/svg-flatten'
+    '';
+
+    installPhase = ''
+      runHook preInstall
+      mkdir -p $out/bin
+      PREFIX=$out make install
+      runHook postInstall
+    '';
+
+    meta = with lib; {
+      description = "svg-flatten SVG downconverter";
+      homepage = "https://github.com/jaseg/gerbolyze";
+      license = with licenses; [ agpl3 ];
+      maintainers = with maintainers; [ wulfsta ];
+    };
+  };
+in pythonPackages.buildPythonApplication rec {
+  inherit pname version src svg-flatten;
+  format = "setuptools";
+
+  nativeBuildInputs = [
+    pythonPackages.setuptools
+  ];
+
+  propagatedBuildInputs = [
+    pythonPackages.beautifulsoup4
+    pythonPackages.click
+    pythonPackages.numpy
+    pythonPackages.scipy
+    pythonPackages.python-slugify
+    pythonPackages.lxml
+    pythonPackages.gerbonara
+    resvg
+    svg-flatten
+  ];
+
+  preConfigure = ''
+    substituteInPlace setup.py \
+      --replace "version = get_version()," \
+                "version = '${version}'," \
+
+    substituteInPlace setup.py \
+      --replace "long_description=format_readme_for_pypi()," \
+                "long_description='\n'.join(Path('README.rst').read_text().splitlines()),"
+  '';
+
+  pythonImportsCheck = [ "gerbolyze" ];
+
+  nativeCheckInputs = [ pythonPackages.pytest resvg svg-flatten ];
+  checkPhase = ''
+    runHook preCheck
+
+    pytest
+
+    runHook postCheck
+  '';
+
+  meta = with lib; {
+    description = "Directly render SVG overlays into Gerber and Excellon files ";
+    homepage = "https://github.com/jaseg/gerbolyze";
+    license = with licenses; [ agpl3 ];
```

Comment:
Add `meta.mainProgram`

Reply:
Can you explain why this is needed? It seems like this setting just defines the name of the binary that is executed for `svg-flatten`, and I'm not seeing any reason we need to set that.

Reply:
It's not strictly *needed* but it's expected to be there by `nix run` (see #219567) and some nixpkgs library functions.

Reply:
Done. Do we also need it in `svg-flatten` or is this required for top-level only?

Reply:
For both, just as long as the package has a main binary.